fdmi/fol_fdmi_src.c

```diff
@@ -104,6 +104,96 @@
  * rh_lsn_discarded for its BE domain could be discarded from deduplication
  * persistence because there is nothing in the cluster that is going to send
  * FDMI FOL record about this operation.
+ *
+ * @section FDMI FOL records resend
+ *
+ * There are 2 major cases here:
+ *
+ * 1. FDMI plugin restarts, FDMI source is not. In this case FDMI source needs
+ *    to resend everything FDMI plugin hasn't confirmed consumption for. This
+ *    could be done by FDMI source dock fom by indefinitely resending FDMI
+ *    records until either FDMI plugin confirms consumption or FDMI plugin
+ *    process fails permanently.
+ * 2. FDMI source restarts. The following description is about this case.
+ *
+ * FDMI source may restart unexpectedly (crash/restart) or it might restart
+ * gracefully (graceful shutdown/startup). In either case there may be FDMI FOL
+ * records that don't have consumption confirmation from FDMI plugin. Current
```

Comment:
`consumption confirmation`.
How is this designed/implemented?

1. the FDMI source posts events/notifications to FDMI plugin side. When the events/notifications have been processed, the plugin side will send confirmation back to source side?
2. When the FDMI source side receive such consumption confirmation, what does it do? It should release some reference on the BE tx log? Or it has some persistent metadata to store the "last confirmed LSN"?

Reply:
> How is this designed/implemented?

Slightly incorrectly, unfortunately. See https://github.com/Seagate/cortx-motr/blob/main/fdmi/fdmi.c#L495 for details.

1. It should. If we want to make FDMI record delivery reliable the confirmation must be sent only after the record becomes persistent (in whatever persistent way it's supposed to). If the confirmation is sent before then the record could be silently lost in case of transient failures.
2. `m0_be_tx_put()` because `m0_be_tx_get()` is enough to prevent BE tx discard from BE log.